src/VisualStudio/TestUtilities/Remoting/WorkspaceWrapper.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using RoslynProject = Microsoft.CodeAnalysis.Project;
+using RoslynWorkspace = Microsoft.CodeAnalysis.Workspace;
+
+namespace Roslyn.VisualStudio.Test.Utilities.Remoting
+{
+    internal class WorkspaceWrapper : MarshalByRefObject
+    {
+        private RoslynWorkspace _workspace;
+
+        public WorkspaceWrapper(RoslynWorkspace workspace)
+        {
+            _workspace = workspace;
+        }
+
+        public void AddReference(string projectName, string referenceAssemblyPath)
```

Comment:
An exception will get thrown if `MetadataReference.CreateFromFile` or `RoslynProject.AddMetadataReference` fails, but how do we let the caller know if `TryApplyChanges` doesn't succeed?